book/AIDO/70_developers/50_define.md

```diff
@@ -1,14 +1,90 @@
+
+The next part is the state-machine that defines how we should move step to step. In our case it is defined as follows.
+
+```
+transitions:
+  - [START, success, step1]
+  - [step1, success, SUCCESS]
+  - [step1, failed, FAILED]
+  - [step1, error, ERROR]
+```
+
+We always start at START. On each row we first define the step we begin at, then we define the action, in our case we can have success \(step is successfull\) which leads to the terminal state of SUCCESS. Similarily for failed and error. 
+
+If you have only one step, as the simple regression test here, then you don't have to change this boilerplate.
+
+#### The evaluation
+
+The evaluation is found in the /evaluation folder. The important, non-boilerplate, file is the eval.py. An example of the content of this file can be found bellow.
+
+```
+#!/usr/bin/env python
+import logging
+import math
+
+from duckietown_challenges import wrap_evaluator, ChallengeEvaluator, InvalidSubmission
+
+logging.basicConfig()
+logger = logging.getLogger('evaluator')
+logger.setLevel(logging.DEBUG)
+
+
+class Evaluator(ChallengeEvaluator):
+
+    def prepare(self, cie):
+        cie.set_challenge_parameters({'dummy':1})
+
+    def score(self, cie):
+        solution_output = cie.get_solution_output_dict()
+	'''
+	Define here how scoring of a solution is done.
+	'''
+	temp = solution_output['data']
+	score = 2*temp
+        cie.set_score('score1', score, 'blurb')
+
+
+if __name__ == '__main__':
+    wrap_evaluator(Evaluator())
+```
+
+What we care about here is the prepare and the score functions. In the prepare function we can define the input we want to give the users submission code. The simplest way to do this is to define a dictionary containing fields for all the input data and then pass this dictionary to the cie.set\_challenge\_parameters function. This dictionary can then be accessed in the solution code by calling cie.get\_challenge\_parameters. The second part is the score function. This function takes the contents of you put in the solution\_output\_dict in your submission and uses it to generate a numerical score. The numerical score is then passed to the system by calling cie.set\_score with the corresponding score field (since we only have one score we called this score1 in the challenge.yaml file) along with the actual numerical score we want to give to the submission and a text blurb related to the score.
```

Comment:
Things like this:

     cie.set\_challenge\_parameters 

should be

    `cie.set_challenge_parameters`